flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -403,6 +405,32 @@ public void testBoundedBuffer() throws Exception {
 		assertEquals(1, localBufferPool.getNumberOfAvailableMemorySegments());
 	}
 
+	@Test
+	public void testIsAvailable() throws Exception {
+
+		// the local buffer pool should be in available state initially
+		assertTrue(localBufferPool.isAvailable().isDone());
+
+		// request one buffer
+		final Buffer buffer = checkNotNull(localBufferPool.requestBuffer());
+		assertFalse(localBufferPool.isAvailable().isDone());
+
+		// set the pool size
+		localBufferPool.setNumBuffers(2);
+		assertTrue(localBufferPool.isAvailable().isDone());
+
+		// reset the pool size
+		localBufferPool.setNumBuffers(1);
+		assertFalse(localBufferPool.isAvailable().isDone());
+
+		final CompletableFuture<?> availableFuture = localBufferPool.isAvailable();
```

Comment:
nit: it is better to put it together either with above or below for readable. E.g.

```
// reset the pool size
localBufferPool.setNumBuffers(1);
final CompletableFuture<?> availableFuture = localBufferPool.isAvailable();
assertFalse(availableFuture.isDone());
```